### PR TITLE
Update recipes for Llama 3 8b and 70b

### PR DESCRIPTION
**Pull request by hemildesai**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/gpt/model/llama.py [hunk 1]
```diff
@@ -59,15 +59,46 @@ class Llama2Config70B(LlamaConfig):
...
+    seq_length: int = 8192
+    num_layers: int = 32
+    hidden_size: int = 4096
     ffn_hidden_size: int = 14336
+    num_attention_heads: int = 32
 
 
 @dataclass
-class Llama3Config70B(Llama2Config70B):
+class Llama3Config70B(Llama3Config):
```
Comment (maanug-nv) [inline nemo/collections/llm/gpt/model/llama.py:94]: should we include the long context configs as well?

File: nemo/collections/llm/recipes/optim/adam.py [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
+from nemo.collections.llm.utils import Config
+from nemo.lightning.pytorch.optim import CosineAnnealingScheduler, MegatronOptimizerModule, OptimizerModule
 
 
-@factory
-def adam_with_cosine_annealing() -> nl.OptimizerModule:
-    return nl.MegatronOptimizerModule(
-        config=OptimizerConfig(optimizer="adam", lr=0.001, use_distributed_optimizer=True),
-        lr_scheduler=nl.lr_scheduler.CosineAnnealingScheduler(),
+def distributed_fused_adam_with_cosine_annealing() -> Config[OptimizerModule]:
```
Comment (maanug-nv) [inline nemo/collections/llm/recipes/optim/adam.py:8]: some of these don't match https://gitlab-master.nvidia.com/dl/JoC/nemo-ci/-/issues/513

File: nemo/collections/llm/recipes/optim/adam.py [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
+    opt_cfg = Config(
+        OptimizerConfig,
+        optimizer="adam",
+        lr=0.0001,
+        weight_decay=0.1,
+        bf16=True,
+        params_dtype=torch.bfloat16,
```
Comment (maanug-nv) [inline nemo/collections/llm/recipes/optim/adam.py:15]: we should remove this, it does nothing in MCore

File: nemo/collections/llm/recipes/optim/adam.py (hunk 2 above)
Comment (hemildesai) [reply nemo/collections/llm/recipes/optim/adam.py]: Updated in https://github.com/NVIDIA/NeMo/pull/10147/commits/d6fd76076addb91fefb356b1de0494539c65212a

File: nemo/collections/llm/recipes/optim/adam.py (hunk 3 above)
Comment (hemildesai) [reply nemo/collections/llm/recipes/optim/adam.py]: Updated in https://github.com/NVIDIA/NeMo/pull/10147/commits/d6fd76076addb91fefb356b1de0494539c65212a

File: nemo/collections/llm/recipes/__init__.py [hunk 4]
```diff
@@ -1,13 +1,14 @@
-from nemo.collections.llm.recipes import llama2_7b, llama3_8b, llama3_8b_16k, llama3_8b_64k, mistral
-from nemo.collections.llm.recipes.log.default import default_log
+from nemo.collections.llm.recipes import llama3_8b, llama3_8b_16k, llama3_8b_64k, llama3_70b, mistral
+from nemo.collections.llm.recipes.log.default import default_log, default_resume
 from nemo.collections.llm.recipes.optim import adam
 
 __all__ = [
     "llama3_8b",
     "llama3_8b_16k",
```
Comment (ShriyaPalsamudram) [inline nemo/collections/llm/recipes/__init__.py:7]: Do we need llama3_8b_16? and llama3_8b_64k? 
We should drop them and maybe add only llama3_70b_64k?

File: nemo/collections/llm/recipes/llama3_70b.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    trainer = Config(
+        nl.Trainer,
+        accelerator="gpu",
+        accumulate_grad_batches=1,
+        callbacks=callbacks,
+        devices=num_gpus_per_node,
+        gradient_clip_val=1.0,
+        limit_test_batches=50,
+        limit_val_batches=32,
+        log_every_n_steps=10,
```
Comment (ShriyaPalsamudram) [inline nemo/collections/llm/recipes/llama3_70b.py:66]: Default to 1 so steps are not skipped?

File: nemo/collections/llm/recipes/__init__.py (hunk 4 above)
Comment (BoxiangW) [reply nemo/collections/llm/recipes/__init__.py:7]: They are part of the long context recipe so I think we should keep them

Comment (BoxiangW) [reply nemo/collections/llm/recipes/__init__.py:7]: I can use another separate PR to address this. https://github.com/NVIDIA/NeMo/pull/10140

File: nemo/collections/llm/recipes/llama3_70b.py (hunk 5 above)
Comment (hemildesai) [reply nemo/collections/llm/recipes/llama3_70b.py:66]: Yeah I can change it, but could become a bit verbose. Also not sure if it will have any impact on performance.

Comment (ShriyaPalsamudram) [reply nemo/collections/llm/recipes/llama3_70b.py:66]: Since nemo 1.0 also uses 10, makes sense to skip steps and reduce verbosity. Thanks

**Review by ShriyaPalsamudram — APPROVED** [review]

**Review by ShriyaPalsamudram — APPROVED** [review]
